pkgs/applications/blockchains/liana/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, llvmPackages
+, fetchurl
+, pkg-config
+, copyDesktopItems
+, makeDesktopItem
+, systemd
+, cmake
+, pkgs
+, makeWrapper
+, xorg
+, expat
+, fontconfig
+, freetype
+, libGL
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "liana";
+  version = "1.1";
+
+  src = fetchFromGitHub {
+    owner = "wizardsardine";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-cqvjPa/GXDdFv8XEPJgHfKP/Duzv+jmp1yZ1rlKPfE8=";
+  };
+
+  cargoLock = {
+    lockFile = "${src}/gui/Cargo.lock";
+    outputHashes = {
+      "liana-1.1.0" = "sha256-ncy7/w9kqD3w+/v0OVKgtgCHCF7Ram327I8R6IJe/s8=";
+      "miniscript-9.0.0" = "sha256-5ITjVqh5frqjTTs993xqo5WwDZ5dDAfFn4DGqt2QbiY=";
+    };
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  buildInputs = [
+    expat
+    fontconfig
+    freetype
+    freetype.dev
+    libGL
+    pkg-config
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXi
+    xorg.libXrandr
+    systemd
+  ];
+
+  sourceRoot = "source/gui";
+
+  postInstall = ''
+    install -Dm0644 $src/gui/ui/static/logos/liana-app-icon.svg $out/share/icons/hicolor/scalable/apps/liana.svg
```

Comment:
Can you try to strip off `$src/gui/` ? The reason is because it is already unpacked in the build directory and that makes working with `nix-shell` a bit easier.